src/radio-button/radio-button.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import {IconFilledCircle} from '../icons';
+import {RadioButtonProps, RadioButtonIconProps} from './types';
+import defaults from './defaults';
+import stylePresets from './style-presets';
+import {withOwnTheme} from '../utils/with-own-theme';
+import {SwitchBase} from '../switch-base';
+import {useRadioGroup} from './radio-group';
+import {composeEventHandlers} from '../utils/compose-event-handlers';
+
+const DefaultIcon = ({checked, iconSize, overrides}: RadioButtonIconProps) =>
+  checked ? (
+    <IconFilledCircle
+      overrides={{
+        size: iconSize,
+        ...overrides,
+      }}
+    />
+  ) : null;
```

Comment:
```suggestion
  checked &&
    <IconFilledCircle
      overrides={{
        size: iconSize,
        ...overrides,
      }}
    />
```

Something like that should work and we do not need any ternary. Not sure if needs to be wrapped in a { }.

Reply:
it will not work since that will return undefined when checked is false, that is not valid react component